flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/BinaryString.java

```diff
@@ -1242,599 +763,32 @@ private void skipBytes(int n, int segSize) {
-	 *  {@code null} splits on whitespace
-	 * @return an array of parsed Strings, {@code null} if null String was input
-	 * @since 2.4
-	 */
-	public BinaryString[] splitByWholeSeparatorPreserveAllTokens(BinaryString separator) {
-		ensureMaterialized();
-		final int len = sizeInBytes;
-
-		if (len == 0) {
-			return EMPTY_STRING_ARRAY;
-		}
-
-		if (separator == null || EMPTY_UTF8.equals(separator)) {
-			// Split on whitespace.
-			return splitByWholeSeparatorPreserveAllTokens(fromString(" "));
-		}
-		separator.ensureMaterialized();
-
-		final int separatorLength = separator.sizeInBytes;
-
-		final ArrayList<BinaryString> substrings = new ArrayList<>();
-		int beg = 0;
-		int end = 0;
-		while (end < len) {
-			end = SegmentsUtil.find(
-				segments, offset + beg, sizeInBytes - beg,
-				separator.segments, separator.offset, separator.sizeInBytes) - offset;
-
-			if (end > -1) {
-				if (end > beg) {
-
-					// The following is OK, because String.substring( beg, end ) excludes
-					// the character at the position 'end'.
-					substrings.add(BinaryString.fromAddress(segments, offset + beg, end - beg));
-
-					// Set the starting point for the next search.
-					// The following is equivalent to beg = end + (separatorLength - 1) + 1,
-					// which is the right calculation:
-					beg = end + separatorLength;
-				} else {
-					// We found a consecutive occurrence of the separator.
-					substrings.add(EMPTY_UTF8);
-					beg = end + separatorLength;
-				}
-			} else {
-				// String.substring( beg ) goes from 'beg' to the end of the String.
-				substrings.add(BinaryString.fromAddress(segments, offset + beg, sizeInBytes - beg));
-				end = len;
-			}
-		}
-
-		return substrings.toArray(new BinaryString[0]);
-	}
-
-	/**
-	 * Calculate the hash value of a given string use {@link MessageDigest}.
-	 */
-	public BinaryString hash(MessageDigest md) {
-		String str = EncodingUtils.hex(md.digest(getBytes()));
-		return fromString(str);
-	}
-
-	public BinaryString hash(String algorithm) throws NoSuchAlgorithmException {
-		return hash(MessageDigest.getInstance(algorithm));
-	}
-
-	private static final List<BinaryString> TRUE_STRINGS =
-		Stream
-			.of("t", "true", "y", "yes", "1")
-			.map(BinaryString::fromString)
-			.peek(BinaryString::ensureMaterialized)
-			.collect(Collectors.toList());
-
-	private static final List<BinaryString> FALSE_STRINGS =
-		Stream
-			.of("f", "false", "n", "no", "0")
-			.map(BinaryString::fromString)
-			.peek(BinaryString::ensureMaterialized)
-			.collect(Collectors.toList());
-
-	/**
-	 * Decide boolean representation of a string.
+	 * Returns the number of bytes for a code point with the first byte as `b`.
+	 * @param b The first byte of a code point
 	 */
-	public Boolean toBooleanSQL() {
-		if (TRUE_STRINGS.contains(toLowerCase())) {
-			return true;
-		} else if (FALSE_STRINGS.contains(toLowerCase())) {
-			return false;
+	static int numBytesForFirstByte(final byte b) {
```

Comment:
move this to `BinaryStringUtil`?

Reply:
To avoid `BinaryString`'s dependence on `BinaryStringUtil`, I think it's more appropriate to put it in `BinaryString`.